.github/workflows/gotest.yml

```diff
@@ -0,0 +1,73 @@
+name: Go Test
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  build_and_test:
+    name: Build and Test
+    strategy:
+      matrix:
+        include:
+          - arch: [x64, qemu-host]
+            image: ghcr.io/viamrobotics/canon:amd64-cache
+            platform: linux/amd64
+            lint: 'make lint'
+          - arch: [arm64, qemu-host]
+            image: ghcr.io/viamrobotics/canon:arm64-cache
+            platform: linux/arm64
+            lint: 'make lint'
```

Comment:
[nit] I don't think we need to define 'lint' here, since it's the same for both platforms.

Reply:
Does this mean I would remove this line and line 16, and do the following?

Change this:
```yml
    - name: Verify no uncommitted changes from "${{ matrix.lint }}"
      run: |
        git init
        git add .
        chown -R testbot .
        sudo -u testbot bash -lc '${{ matrix.lint }}'
        GEN_DIFF=$(git status -s)
```

to this?

```yml
    - name: Verify no uncommitted changes from "make lint"
      run: |
        git init
        git add .
        chown -R testbot .
        sudo -u testbot bash -lc make lint
        GEN_DIFF=$(git status -s)
```

Reply:
yes

Reply:
@Otterverse Do I have to put `make lint` in between ticks?

Referring to that, the above works. Specifically, this part: 
```yml
sudo -u testbot bash -lc make lint
```

Whereas for the tests, we have this:
```yml
    - name: Test
      run: |
        sudo --preserve-env=MONGODB_TEST_OUTPUT_URI,GITHUB_SHA,GITHUB_RUN_ID,GITHUB_RUN_NUMBER,GITHUB_RUN_ATTEMPT,GITHUB_X_PR_BASE_SHA,GITHUB_X_PR_BASE_REF,GITHUB_X_HEAD_REF,GITHUB_X_HEAD_SHA,GITHUB_REPOSITORY -u testbot bash -lc 'make test'
```

where we have `'make test'` instead of `make test`. Why and when would I put `make lint` and `make test` between ticks?

Reply:
So generally, you need to quote the string passed to bash with the -c option. You can test this yourself in the command line.

`bash -c echo fotbar` returns nothing.
`bash -c "echo foobar"` or `bash -c 'echo foobar'` should both actually print out "foobar"

(The discussion over WHICH quote marks to use is another topic, but for simple things with no special characters, either works.)

If make lint was working without the quotes, then it's likely JUST running "make" and not actually passing the second word. That might still work in your case, if the default "make" thing does the same stuff (or does nothing, and thus nothing changes and it still passes the test.)

Reply:
Great, good to know! Glad I asked :) Will fix now.